lib/stripe/plans.rb

```diff
@@ -54,6 +63,23 @@ def name_or_product_id
         errors.add(:base, 'must have a product_id or a name') unless (@product_id.present? ^ @name.present?)
       end
 
+      def billing_scheme_must_be_tiered
+        errors.add(:billing_scheme, 'must be set to `tiered` when specifying `tiers`') unless billing_scheme == 'tiered'
+      end
+
+      def tiers_must_be_array
+        errors.add(:tiers, 'must be an Array') unless tiers.is_a?(Array)
+      end
+
+      def set_tiers
+        @tiers = tiers.map { |t| t.is_a?(Stripe::Plans::BillingTier) ? t : Stripe::Plans::BillingTier.new(t) } if tiers
```

Comment:
I'm worried about overwriting the existing tiers instance variable. It's probably not thread-safe and also the content of `@tiers` would be determined by when `set_tiers` is called. Maybe use another variable? e.g.

```ruby
def billing_tiers
  @billing_tiers ||= tiers.map { |t| Stripe::Plans::BillingTier.new(t) }
end

def validate_tiers
  billing_tiers.all?(&:valid?)
end
```